src/app/_components/Pagination.tsx

```diff
@@ -13,7 +17,13 @@ export function Pagination({
   currentPage,
   setCurrentPage,
 }: PaginationProps) {
-  const pageNumbers = Array.from({ length: pageCount }, (_, index) => index + 1)
+  const isXsDevice = useMediaQuery('only screen and (max-width : 640px)')
+
+  const collapsedRange = useCollapseRange(
+    pageCount,
+    currentPage,
+    isXsDevice ? 2 : 6,
```

Comment:
The `2` and `6` are like[ magic numbers](https://www.inf.unibz.it/~calvanese/teaching/05-06-ip/lecture-notes/uni04/node17.html#:~:text=A%20magic%20number%20is%20a,difficult%20to%20maintain%20and%20update.) - We can define them with variables.

Reply:
Love it! 

I added a `useResponsiveRange` to compute the range's size based on the screen size. Hopefully, the numbers make more sense in that context.